screen-search/src/main/java/com/ivy/search/SearchViewModel.kt

```diff
@@ -29,48 +24,52 @@ class SearchViewModel @Inject constructor(
     private val categoriesAct: CategoriesAct,
     private val baseCurrencyAct: BaseCurrencyAct,
     private val allTrnsAct: AllTrnsAct
-) : ViewModel() {
+) : ComposeViewModel<SearchState, SearchEvent>() {
 
-    private val _baseCurrencyCode = MutableStateFlow(getDefaultFIATCurrency().currencyCode)
-    val baseCurrencyCode = _baseCurrencyCode.asStateFlow()
+    private val _uiState = mutableStateOf(
+        SearchState(
+            transactions = persistentListOf(),
+            baseCurrency = getDefaultFIATCurrency().currencyCode,
+            accounts = persistentListOf(),
+            categories = persistentListOf()
+        )
+    )
 
-    private val _transactions =
-        MutableStateFlow<ImmutableList<TransactionHistoryItem>>(persistentListOf())
-    val transactions = _transactions.asStateFlow()
-
-    private val _accounts = MutableStateFlow<ImmutableList<Account>>(persistentListOf())
-    val accounts = _accounts.asStateFlow()
+    @Composable
+    override fun uiState(): SearchState {
+        return _uiState.value
```

Comment:
Instead of having one big `uiState` mutableStateOf create a couple of small ones:
- transactions
- accounts
- categories

and here do
```
    return SearchState(
       transactions = transactions.value, 
        ...
    )
```